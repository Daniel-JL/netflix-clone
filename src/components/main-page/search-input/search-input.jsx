/* eslint-disable import/prefer-default-export */
import React, { useState } from 'react';
import {
  NavLink,
  useLocation,
  useHistory,
} from 'react-router-dom';
import styled from 'styled-components';

const SearchInput = () => {
  const location = useLocation();
  const history = useHistory();
  const [searchInputText, setSearchInputText] = useState(`${location.pathname}`);
  const [initialUrl] = useState(`${location.pathname}`);

  const handleChange = (e) => {
    setSearchInputText(`${e.target.value}`);
    if (e.target.value === '') {
      history.push(initialUrl);
    } else {
      history.push(`search?q=${e.target.value}`);
    }
  };

  return (
    <div>
      <NavLink to={`${searchInputText}`}>
        <input
          type="text"
          onChange={(e) => handleChange(e)}
        />
      </NavLink>
    </div>
  );
};

export default SearchInput;
