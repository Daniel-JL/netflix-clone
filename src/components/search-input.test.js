/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import {
  MemoryRouter,
  Router,
} from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { SearchInput } from './search-input';

describe('Search input', () => {
  it('should render correctly', () => {
    const wrapper = renderer.create(
      <MemoryRouter>
        <SearchInput />
      </MemoryRouter>,
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should cause app to enter search view when something is typed in bar', () => {
    const history = createMemoryHistory();
    const { getByRole } = render(
      <Router history={history}>
        <SearchInput />
      </Router>,
    );
    const textInput = getByRole('textbox', { name: '' });

    userEvent.type(textInput, 'a');
    expect(history.location.pathname).toBe('/search');
    expect(history.location.search).toBe('?q=a');

  });

  it('should add text to url as it is input', () => {
    const history = createMemoryHistory();
    const { getByRole } = render(
      <Router history={history}>
        <SearchInput />
      </Router>,
    );
    const textInput = getByRole('textbox', { name: '' });

    userEvent.type(textInput, 'a');
    expect(history.location.pathname).toBe('/search');
    expect(history.location.search).toBe('?q=a');
  });

  it('should remember initial url state', () => {

  });

  it('should add an x button to the box when something is typed. This button clears the search ', () => {

  });

});
