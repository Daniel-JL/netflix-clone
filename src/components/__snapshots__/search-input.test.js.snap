// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search input should render correctly 1`] = `
<div>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <input
      onChange={[Function]}
      type="text"
    />
  </a>
</div>
`;
